pkg/controller/core.oam.dev/v1alpha2/application/apply.go

```diff
@@ -231,7 +232,8 @@ func (h *AppHandler) collectHealthStatus(ctx context.Context, wl *appfile.Worklo
 	if wl.CapabilityCategory == types.TerraformCategory {
 		var configuration terraformapi.Configuration
 		if err := h.r.Client.Get(ctx, client.ObjectKey{Name: wl.Name, Namespace: namespace}, &configuration); err != nil {
-			return nil, false, errors.WithMessagef(err, "app=%s, comp=%s, check health error", appName, wl.Name)
+			klog.ErrorS(err, "failed to get configuration", "app name", appName, "comp name", wl.Name)
+			return nil, false, nil
```

Comment:
why don't return the error here?

Reply:
Same question as @wonderflow . Maybe I am missing what the fix here is as it only seems to log the error.

Reply:
If we return the error here, in the application status, we can see the `configuration not found error...`. but this error is caused by we failed to get configuration when we check the health.

Reply:
the root cause is why we will miss the configuration here? Will we periodically check the existence of configuration?